### ISSUE TITLE
Add youtube and vimeo link check support

### ISSUE THREAD

**Issue by pawroman** [lycheeverse/lychee#28]
For youtube links, it's necessary to scrape the content of the website to work out if the video still exists. For vimeo, they don't really like fake user agents.

The https://www.youtube.com/oembed API will return 404 for missing videos and can be used to check youtube links.

We special case GitHub already, we could special case YouTube and Vimeo too. The mechanism to specialize these kinds of links should be made extensible though.

**Comment by mre** [comment]
@pawroman and me discussed that we could have a `workarounds` or `overwrites` module for these cases.
Perhaps a structure like this would work:
```rust
struct Overwrite {
  match: regex::Regex,
  user_agent: String,
  headers: HashSet<Header>,
  // ...
}
```
We could think about more generic approaches, but I'd be careful not to generalize too early and only add the overwrites we need in the beginning. Any ideas?

**Comment by mre** [comment]
I'm working on this in #156.

Tested vimeo and it seems to be okay with a fake user agent (?):

```
 curl -sIXGET  -A "totally-legit-useragent" https://vimeo.com/510814675
 ```

This gives me a 200.

```
 curl -sIXGET  -A "totally-legit-useragent" https://vimeo.com/510814675invalid
 ```

This gives me a 404 as expected.

Perhaps they changed something on their end?

**Comment by mre** [comment]
This should be fixed now in `master` because #156 is merged. 🎉
